engine/graphics/src/vulkan/graphics_vulkan_private.h

```diff
@@ -309,18 +302,37 @@ namespace dmGraphics
         const VulkanResourceType GetType();
     };
 
+    struct VulkanCommandBuffer
+    {
+        struct VulkanHandle
+        {
+            VkCommandBuffer m_CommandBuffer;
+            VkCommandPool   m_CommandPool;
+        };
+
+        VulkanHandle m_Handle;
+    };
+
     struct ResourceToDestroy
     {
         union
         {
-            DeviceBuffer::VulkanHandle  m_DeviceBuffer;
-            VulkanTexture::VulkanHandle m_Texture;
-            VulkanProgram::VulkanHandle m_Program;
-            RenderTarget::VulkanHandle  m_RenderTarget;
+            DeviceBuffer::VulkanHandle        m_DeviceBuffer;
+            VulkanTexture::VulkanHandle       m_Texture;
+            VulkanProgram::VulkanHandle       m_Program;
+            RenderTarget::VulkanHandle        m_RenderTarget;
+            VulkanCommandBuffer::VulkanHandle m_CommandBuffer;
         };
         VulkanResourceType m_ResourceType;
     };
 
+    struct FenceResourcesToDestroy
+    {
+        VkFence           m_Fence;
+        ResourceToDestroy m_Resources[MAX_FENCE_RESOURCES_TO_DESTROY_PER_ENTRY];
+        uint8_t           m_ResourcesCount;
+    };
```

Comment:
It's intentionally generic, we can use the same syncronization structure for other types of CPU -> GPU uploads. E.g vertex/index buffer uploads.